### ISSUE TITLE
MarkLogic forces Java developers to use Spring

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic/marklogic-data-hub#1704]
MarkLogic is forcing Java developers to use spring since 4.1.0. 

For example, `FlowManagerImpl` has no constructor/setter for `HubConfig`.  This fully relies on Springs auto-wiring facility (see below).

It is nice to use spring, but the DHF API should not enforce it, especially if it is not really necessary (in this case)

I have my own HubConfig class (because I want to run this without project directory reference) which supplies DHF with all info. However without Spring I cannot inject this because there is no setter or constructor (I am aware I can inject via reflection). 

So my workaround for now is to copy entire `FlowManagerImpl` and add a constructor to get this done.

```
public class FlowManagerImpl extends ResourceManager implements FlowManager {
 
    @Autowired
    private HubConfig hubConfig;
```

**Comment by srinathgit** [comment]
Please check out this example https://marklogic.github.io/marklogic-data-hub/harmonize/java/. 'HubConfig' has a singleton scope starting 4.1.0. Once spring creates the objects , you can set the values for HubConfig properties using the setter methods and call hubconfig.refreshProject(), FlowManager would use the updated HubConfig .

**Comment by josvanroosmalen** [comment]
Yes but this still requires spring.

**Comment by srinathgit** [comment]
Yes, you are correct, it would require Spring.

**Comment by josvanroosmalen** [comment]
If you make HubConfig a POJO will this fix issue #1521 ? That is issue is related that FlowRunner need to have access to the whole project directory. I fixed this by: making a new class, implement HubConfig, throw by default for every method (there are a lot of methods) a RuntimeException. Then I ran FlowRunner, to see which method where hit (not a lot) and implemented them.

**Comment by rjrudin** [comment]
Closing this, as 5.2.0 will include new constructors for FlowRunnerImpl so that you can run a flow without requiring Spring and without requiring project files. I believe the only reason you'd need FlowManagerImpl is because of the dependency on project files, but that's no longer a dependency.

While I think it would be useful for other DHF classes to have constructors that make Spring optional, I think the need for that should be driven via future tickets that identify the use case.
